Add Fuel cycle benchmarking (#234)
idaholab/TMAP8: doc/content/verification_and_validation/fuel_cycle_benchmarking.md

# Fuel Cycle Benchmarking

## General Case Description

This benchmarking case is taken from [!cite](meschini2023modeling) to simulate a tritium fuel cycle model using resident time method. The model uses a simplified approach by assigning residence times to simulate tritium flow through each system in a fusion power plant, avoiding complex and unfinished technical details. This approach helps understand the challenges and potential solutions for optimizing tritium inventory management and accelerating fusion energy development.

> ## Model Description

Reviewer: Please also specify what the key differences are with the other fuel cycle model.

       caption=Schematic of the tritium fuel cycle model showing the main systems and tritium flow paths. The figure is taken from [!cite](meschini2023modeling).

The label and corresponding equation for each systems are shown in [tritium_systems].

!table id=tritium_systems caption=Systems and labels used in this example.
| System Name | System number | Tritium inventory variable| system equation |
| --- | --- | --- | --- |
| Breeding Blanket                       | 1  | `T_01_BB`       | [eqn:t1] |
| Tritium Extraction System              | 2  | `T_02_TES`      | [eqn:t2] |
| First Wall                             | 3  | `T_03_FW`       | [eqn:t3] |
| Divertor                               | 4  | `T_04_DIV`      | [eqn:t4] |
| Heat Exchanger                         | 5  | `T_05_HX`       | [eqn:t5] |
| Detritiation System                    | 6  | `T_06_DS`       | [eqn:t6] |
| Vaccum Pump                            | 7  | `T_07_vacuum`   | [eqn:t7] |
| Fuel Clean-up                          | 8  | `T_08_FCU`      | [eqn:t8] |
| Isotope Separation System              | 9  | `T_09_ISS`      | [eqn:t9] |
| Storage and Management                 | 10 | `T_10_storage`  | [eqn:t10] |

> | Fueling System                         | 11 | -               | -         |

Reviewer: Why have this system if there is no variable nor equation associated with it?

| $\eta_2$  | Tritium extraction efficiency        | 0.7                                                         | -                     |
| $f_{DIR}$ | Direct internal recycling fraction   | 0.5                                                         | -                     |
| $I_i$     | Tritium inventory in the $i$th component | -                                                       | kg                    |
| $\dot{N}_{T,burn}$ | Tritium burn rate           | 8.99e-7                                                     | -                     |
| $\lambda$ | Tritium decay rate                   | 1.73e-9                                                     | s$^{-1}$              |
| $t$       | time                                 | -                                                           | s                     |
| $\tau_i$  | Tritium residence time in the $i$th component | 4500 in $\tau_1$, 86400 in $\tau_2$, 1000 in $\tau_3$, $\tau_4$, $\tau_5$, 3600 in $\tau_6$, 600 in $\tau_7$, 585 in $\tau_8$, 22815 in $\tau_9$, 100 in $\tau_11$ | s                     |

## Results

> The model is validated by comparing TMAP8 simulation results with MatLab calculations from [!cite](meschini2023modeling) at first 20 days. [fuel_cycle_comparison] shows excellent agreement in the temporal evolution of tritium inventory across key systems, including breeding blanket, tritium extraction, vacuum pump, and storage systems. The close match validates our implementation using MOOSE's ScalarKernel system to solve the coupled ODEs that describe tritium transfer between systems.

Reviewer: ```suggestion
The model is benchmarked by comparing TMAP8 simulation results with MatLab calculations from [!cite](meschini2023modeling) during the first 20 days. [fuel_cycle_comparison] shows excellent agreement in the temporal evolution of tritium inventory across key systems, including breeding blanket, tritium extraction, vacuum pump, and storage systems. The close match properly benchmarks our implementation in TMAP8 using MOOSE's ScalarKernel system to solve the coupled ODEs describing tritium transfer between systems.
```

!media comparison_fuel_cycle_benchmark.py
       image_name=fuel_cycle_comparison.png
       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
       id=fuel_cycle_comparison
       caption=Comparison of TMAP8 calculation with the data of fuel cycle model from [!cite](meschini2023modeling).

## Input files

!style halign=left

> The input file for this case can be found at [/fuel_cycle_benchmark/fuel_cycle.i]. The input file used for testing has fewer time steps to limit computational costs. More information about the changes can be found in the test specification file at [/fuel_cycle_benchmark/tests].

Reviewer: This description is currently not very accurate since you do not just limit the number of time step. Please update.

| $\tau_i$  | Tritium residence time in the $i$th component | 4500 in $\tau_1$, 86400 in $\tau_2$, 1000 in $\tau_3$, $\tau_4$, $\tau_5$, 3600 in $\tau_6$, 600 in $\tau_7$, 585 in $\tau_8$, 22815 in $\tau_9$, 100 in $\tau_11$ | s                     |

## Results

The model is benchmarked by comparing TMAP8 simulation results with MatLab calculations from [!cite](meschini2023modeling) during the first 20 days. [fuel_cycle_comparison] shows excellent agreement in the temporal evolution of tritium inventory across key systems, including breeding blanket, tritium extraction, vacuum pump, and storage systems. The close match properly benchmarks our implementation in TMAP8 using MOOSE's [`ScalarKernel`](/syntax/ScalarKernels) system to solve the coupled ODEs describing tritium transfer between systems.

!media comparison_fuel_cycle_benchmark.py
       image_name=fuel_cycle_comparison.png
       style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
       id=fuel_cycle_comparison
       caption=Comparison of TMAP8 calculation with the data of fuel cycle model from [!cite](meschini2023modeling).

!alert note title=The comparison only considers first 20 days.

> The data file from [!cite](meschini2023modeling) for a three-year simulation has a large storage size due to the short time intervals used. In addition, the trend of tritium curves in these critical systems do not show significant variance after 20 days. To reduce storage costs, we only compare the TMAP8 simulation results with MatLab calculations from [!cite](meschini2023modeling) for the first 20 days.

Reviewer: ```suggestion
The data file from [!cite](meschini2023modeling) for a three-year simulation has a large storage size due to the short time intervals used. In addition, the trend of tritium curves in these critical systems do not show significant variance after 20 days. To reduce storage costs, we only compare the TMAP8 simulation results with MatLab calculations from [!cite](meschini2023modeling) for the first 20 days, even if the TMAP8 simulation is run for a longer period of time.
```

I think we should run the TMAP8 example for longer (unless it takes way too much time). We can still compare only the first 20 days in the figure.
